### ISSUE TITLE
Reg. Getting Gradient Loss values while testing

### ISSUE THREAD

**Issue by ksivakumar** [nikhilbarhate99/Hierarchical-Actor-Critic-HAC-PyTorch#5]
Hi I want to get the gradient loss with respect to input while I am testing. I tried the following in HAC.py after the `next_state, rew, done, _ = env.step(action)` line in the low level policy:

```
state = torch.FloatTensor(state).to(device)
action = torch.FloatTensor(action).to(device)
goal = torch.FloatTensor(goal).to(device)
criticval = self.HAC[0].critic(state, action, goal).detach()
```
with plans of computing a new target_Q in the moment to calculate loss and then gradient. However, I get the error `Dimension out of range (expected to be in range of [-1, 0], but got 1)`. Any reason why this is happening?

**Comment by nikhilbarhate99** [comment]
seems to be a known [PyTorch error](https://discuss.pytorch.org/t/indexerror-dimension-out-of-range-expected-to-be-in-range-of-1-0-but-got-1/54267)
